### PR TITLE
Introduce MeanSquaredError and SoftmaxCrossEntropy as Loss Function

### PR DESCRIPTION
**Pull request by SherlockNoMad**
This PR is a part of ONNX training spec. If you want to see all components in the training spec proposal, please see #2314

### CODE REVIEW

File: onnx/defs/loss/defs.cc [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    OpSchema()
+        .SetDoc(SoftmaxCrossEntropy_ver10_doc)
+        .Attr(
+            "reduction",
+            reduction_doc,
+            AttributeProto::STRING,
+            std::string("mean"))
+        .Input(
+            0,
+            "predictions",
```
Comment (wschin) [inline onnx/defs/loss/defs.cc]: Maybe?
```suggestion
            "scores",
```
Prediction sounds a predicted label.

File: onnx/defs/loss/defs.cc [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            "same shape as labels; otherwise, it is scalar.",
+            "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {}));
+
+static const char* SoftmaxCrossEntropy_ver10_doc =
+    R"DOC(Loss function that measures the softmax cross entropy between
```
Comment (wschin) [inline onnx/defs/loss/defs.cc]: Please add equations to define this opertor's behavior. I feel using equation is the only way to eliminate uncertainty in ONNX (a mathematical standard).

File: onnx/defs/loss/defs.cc [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+#include "onnx/defs/schema.h"
+namespace ONNX_NAMESPACE {
+const char* reduction_doc =
+    "Type of reduction to apply to loss: none, sum, mean(default). "
+    "'none': no reduction will be applied, "
+    "'sum': the output will be summed. "
+    "'mean': the sum of the output will be divided by the number of "
+    "elements in the output.";
+
+static const char* MSE_ver10_doc = R"DOC(Loss function that measures the
```
Comment (wschin) [inline onnx/defs/loss/defs.cc]: Please add equations to define this opertor's behavior. I feel using equation is the only way to eliminate uncertainty in ONNX (a mathematical standard). Notice that L2-norm is only formally defined for `vector` (aka 1-D tensor) but it looks like this operator accepts general tensors.

File: onnx/defs/loss/defs.cc [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    MeanSquaredError,
+    10,
+    OpSchema()
+        .SetDoc(MSE_ver10_doc)
+        .Attr(
+            "reduction",
+            reduction_doc,
+            AttributeProto::STRING,
+            std::string("mean"))
+        .Input(0, "predictions", "The predicted outputs.", "T")
```
Comment (wschin) [inline onnx/defs/loss/defs.cc]:
```suggestion
        .Input(0, "scores", "The predicted outputs.", "T")
```

**Comment by wschin** [comment]
We have a replacement for this PR, so please allow me to close the PR.
https://github.com/onnx/onnx/pull/2573
https://github.com/onnx/onnx/pull/2570
